src/cocaine-app/infrastructure.py

```diff
@@ -688,38 +777,33 @@ def search_history_by_path(self, request):
 
         entries = []
 
-        if not path.startswith('^'):
-            path = '^' + path
-        if not path.endswith('$'):
-            path = path + '$'
-        path = path.replace('*', '.*')
-
         start_idx = 0
         if params.get('last', False):
             start_idx = -1
 
-        for group_id, group_history in self.state.iteritems():
-            for node_set in group_history['nodes'][start_idx:]:
-                ts = node_set['timestamp']
-                for node in node_set['set']:
-
-                    node_path = node[4]
+        group_histories = self.group_history_finder.search_by_node_backend(
+            hostname=host,
+            path=path
+        )
+        for group_history in group_histories:
+            for node_backends_set in group_history.nodes[start_idx:]:
+                for node_backend in node_backends_set.set:
 
-                    if not node_path:
+                    if not node_backend.path:
                         continue
 
-                    if re.match(path, node_path) is not None and node[0] == host:
-                        entries.append((ts, group_id, node_set))
+                    if re.match(path, node_backend.path) is not None and node_backend.hostname == host:
```

Comment:
This has the same reason as https://github.com/yandex/mastermind/pull/13#discussion_r41516603